nest-city-account/src/oauth2/subservices/oauth2-validation.subservice.ts

```diff
@@ -0,0 +1,405 @@
+
+  constructor(private readonly throwerErrorGuard: ThrowerErrorGuard) {}
+
+  /**
+   * Validate authorization request parameters
+   * Used by both /authorize and /continue endpoints
+   *
+   * @param params - The authorization parameters (from query or decoded payload)
+   */
+  validateAuthorizationRequest(params: AuthorizationParams) {
+    const client = this.validateClientId(params.clientId)
+    this.validateRedirectUri(params.redirectUri, client)
+    this.validateScope(params.scope, client)
+    const { codeChallenge, codeChallengeMethod } = this.validatePkceParameters(
+      params.codeChallenge,
+      params.codeChallengeMethod,
+      client.clientId
+    )
+    this.validateResponseType(params.responseType, client, codeChallenge, codeChallengeMethod)
+  }
+
+  private validateClientId(clientId: string | unknown): ClientConfig {
+    if (!clientId || typeof clientId !== 'string' || clientId.length === 0) {
+      this.logger.error('Missing or invalid client_id in authorization request', {
+        hasClientId: !!clientId,
+        clientIdType: typeof clientId,
+      })
+      throw this.throwerErrorGuard.OAuth2AuthorizationException(
+        OAuth2AuthorizationErrorCode.INVALID_REQUEST,
+        `Invalid request: client_id is required`
+      )
+    }
+
+    const client = findClientById(clientId)
+    if (!client) {
+      this.logger.error('Client not found', {
+        clientId: clientId,
+      })
+      throw this.throwerErrorGuard.OAuth2AuthorizationException(
+        OAuth2AuthorizationErrorCode.UNAUTHORIZED_CLIENT,
+        `Unauthorized client: unknown client`
+      )
+    }
+
+    return client
+  }
+
+  private validateRedirectUri(redirectUri: string | unknown, client: ClientConfig): string {
+    if (!redirectUri || typeof redirectUri !== 'string' || redirectUri.length === 0) {
+      this.logger.error('Missing or invalid redirect_uri in authorization request', {
+        clientId: client.clientId,
+        hasRedirectUri: !!redirectUri,
+        redirectUriType: typeof redirectUri,
+      })
+      throw this.throwerErrorGuard.OAuth2AuthorizationException(
+        OAuth2AuthorizationErrorCode.INVALID_REQUEST,
+        `Invalid request: redirect_uri is required`
+      )
+    }
+
+    if (!isRedirectUriAllowed(client, redirectUri)) {
+      this.logger.error('Redirect URI not allowed for client', {
+        clientId: client.clientId,
+        redirectUri: redirectUri,
+      })
+      throw this.throwerErrorGuard.OAuth2AuthorizationException(
+        OAuth2AuthorizationErrorCode.INVALID_REQUEST,
+        `Invalid request: provided redirect URI is not allowed for this client`
+      )
+    }
+
+    return redirectUri
+  }
+
+  private validateScope(scope: string | unknown, client: ClientConfig): string | undefined {
+    if (scope === undefined) {
+      return undefined
+    }
+    if (typeof scope !== 'string' || !areScopesAllowed(client, scope as string)) {
```

Comment:
```suggestion
    if (typeof scope !== 'string' || !areScopesAllowed(client, scope)) {
```

no need to retype here, typescript knows it must be string if it goes to the second branch 😄